bin/server.py

```diff
@@ -64,41 +69,49 @@ async def search_opportunities(
         product_router: ProductRouter,
         search: OpportunityRequest,
         request: Request,
-    ) -> list[Opportunity]:
-        return [o.model_copy(update=search.model_dump()) for o in self._opportunities]
+    ) -> Result[list[Opportunity], Maybe[Exception]]:
```

Comment:
This is a place where I got a type error, and indeed the type signature seems wrong, as we never "maybe" have an exception. How about:

```suggestion
    ) -> Result[list[Opportunity], Exception]:
```

?

Reply:
It shouldn't be possible to get here if the product doesn't exist as there would be no registered endpoint. That is, FastAPI would handle that directly for us because it would have no matching route for said product, thus we don't need to concern ourselves with handling that case. Unless I'm missing something.

Reply:
ok, we should try that then -- i was mostly thinking of how the STAC API handles those endpoints. Either way, I think we want to be consistent on the types for partially defined functions.